File: Saga.Admin.web/src/models/AdminResponseDto.ts
export interface AdminQuestResponseDto {
  questId: number;

  title: string;

  content: string;

  documentationLink: string;

  videoLink: string;
}

export interface AdminChapterResponseDto {
  chapterId: number;
  title: string;

  content: string;

  rewardPoints: number;

  quests: AdminQuestResponseDto[];
}

export interface AdminResponseDto {
  sagaDataId: number;
  name: string;

  category: string;

  description: string;

  completionBonusPoints: number;

  chapters: AdminChapterResponseDto[];
}

File: Saga.API/DTOs/Admin/AdminResponseDto.cs
using Saga.API.Models;

namespace Saga.API.DTOs.Admin
{
    public class AdminResponseDto
    {
        public int SagaDataId { get; set; }

        public string Name { get; set; } = string.Empty;

        public string Category { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public int CompletionBonusPoints { get; set; }

        public List<AdminChapterResponseDto> Chapters { get; set; } = [];
    }
}
